src/core/client/embed/StreamEmbed.ts

```diff
@@ -0,0 +1,125 @@
+import { EventEmitter2 } from "eventemitter2";
+import qs from "query-string";
+
+import {
+  Decorator,
+  withAutoHeight,
+  withClickEvent,
+  withEventEmitter,
+  withIOSSafariWidthWorkaround,
+  withPymStorage,
+  withSetCommentID,
+} from "./decorators";
+import onIntersect from "./onIntersect";
+import PymControl, {
+  defaultPymControlFactory,
+  PymControlFactory,
+} from "./PymControl";
+import { ensureEndSlash } from "./utils";
+
+export interface StreamEmbedConfig {
+  assetID?: string;
+  assetURL?: string;
+  commentID?: string;
+  autoRender?: boolean;
+  title: string;
+  eventEmitter: EventEmitter2;
+  id: string;
+  rootURL: string;
+}
+
+export class StreamEmbed {
+  private config: StreamEmbedConfig;
+  private pymControl?: PymControl;
+  private pymControlFactory: PymControlFactory;
+
+  constructor(
+    config: StreamEmbedConfig,
+    pymControlFactory = defaultPymControlFactory
+  ) {
+    this.config = config;
+    this.pymControlFactory = pymControlFactory;
+    if (config.commentID) {
+      setTimeout(() => config.eventEmitter.emit("showPermalink"), 0);
```

Comment:
Can you add a comment as to why this is a timeout of 0?

Reply:
Sure!